apps/src/templates/amazonFutureEngineerEligibility/amazonFutureEngineerEligibility.jsx

```diff
@@ -30,27 +30,60 @@ export default class AmazonFutureEngineerEligibility extends React.Component {
   constructor(props) {
     super(props);
 
-    let sessionEligibilityData =
-      JSON.parse(sessionStorage.getItem(sessionStorageKey)) || {};
+    let sessionEligibilityData = this.getSessionEligibilityData();
 
+    // Initial state is set by information that has been stored to the session.
+    // If none exists (ie, a user's first time visiting the page),
+    // set defaults that will ask them to provide school information.
     this.state = {
       formData: {
+        ...sessionEligibilityData,
         signedIn: this.props.signedIn,
-        schoolEligible:
-          sessionEligibilityData.schoolEligible ||
-          this.props.schoolEligible ||
-          null,
+        schoolEligible: this.checkInitialSchoolEligibility(
+          sessionEligibilityData
+        ),
         schoolId:
           sessionEligibilityData.schoolId || this.props.schoolId || null,
-        consentAFE: sessionEligibilityData.consentAFE || false,
-        submitted: sessionEligibilityData.submitted || false
+        consentAFE: sessionEligibilityData.consentAFE || false
       },
       errors: {}
     };
   }
 
-  updateFormData = change => {
-    this.setState({formData: {...this.state.formData, ...change}});
+  // If a user has gone through the eligibility flow,
+  // or has school information associated with their account,
+  // we use that for determining their eligibilty.
+  // If the school information associated with their account
+  // is ineligible, we still allow them to provide their school information
+  // in case their account information is out of date.
+  // Redirect to ineligible page ('/afe/benefits') if ineligible.
+  // Otherwise, we ask them for their school information.
+  checkInitialSchoolEligibility = sessionEligibilityData => {
+    if (
+      sessionEligibilityData.schoolEligible === true ||
```

Comment:
is schoolEligible always a boolean? If so don't need `=== true` or `=== false`, can do `sessionEligibilityData.schoolEligible` or `!sessionEligibilityData.schoolEligible`

Reply:
`schoolEligible` being null is how I've been determining that we should show you the "select a school", so I think I have to compare exactly to true/false?

Reply:
for true you can leave out `=== true` since null will be falsy

Reply:
Updated!